proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/TranslatorApplication.java

```diff
@@ -30,24 +26,26 @@
 import uk.gov.ida.notification.shared.istio.IstioHeaderStorage;
 import uk.gov.ida.notification.shared.logging.ProxyNodeLoggingFilter;
 import uk.gov.ida.notification.shared.proxy.MetatronProxy;
-import uk.gov.ida.notification.shared.proxy.ProxyNodeJsonClient;
+import uk.gov.ida.notification.shared.proxy.MetatronProxyFactory;
 import uk.gov.ida.notification.shared.proxy.VerifyServiceProviderProxy;
 import uk.gov.ida.notification.translator.configuration.TranslatorConfiguration;
 import uk.gov.ida.notification.translator.resources.HubResponseTranslatorResource;
 import uk.gov.ida.notification.translator.saml.EidasFailureResponseGenerator;
 import uk.gov.ida.notification.translator.saml.EidasResponseGenerator;
 import uk.gov.ida.notification.translator.saml.HubResponseTranslator;
 
-import javax.ws.rs.client.Client;
-
 public class TranslatorApplication extends Application<TranslatorConfiguration> {
 
+    private MetatronProxy espMetatronProxy;
```

Comment:
We certainly don't need two metatron proxies here - one is enough and totally fine. The translator only needs one instance - which will be used also by objects it creates.
Also the instance called `espMetatronProxy` here is NOT used by ESP - it's used by `EidasResponseGenerator` which is an object used by the translator. ESP is a separate microservice which is created by its own Application class - it's completely separate from the Translator.

Reply:
Thanks 👍